### PR TITLE
Add notebook and scripts for collecting data for CP1

### PR DESCRIPTION
**Pull request by danlamanna**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -54,3 +54,6 @@ errors.log
 /TPL/libsvm-3.1-custom/svm-prob
 /TPL/libsvm-3.1-custom/svm-scale
 /TPL/libsvm-3.1-custom/svm-train
+/bin/memex/hackathon_2016_07/scripts/es_config.json
```
Comment (Purg) [inline .gitignore]: Check in a config with bogus credentials / paths for example?

File: bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        elif i == NUM_RETRIES:
+            sys.exit(2)
+        elif not r.ok:
+            i += 1
+            continue
+        else:
+            sha1 = hashlib.sha1(r.content).hexdigest()
+
+            try:
+                os.makedirs(os.path.join(DATA_DIR, 'CP1_imageset', sha1[0:3]))
```
Comment (Purg) [inline bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py]: There is a function in ``smqtk.utils.file_utils`` called ``save_create_dir`` that is for this.

File: bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# input is like:
+# http://someimageurl docid1 docid2.. docidn
+
+# Download someimageurl to a location based on its sha1 hash
+# Then print out the mapping of the URL to the sha
+
+NUM_RETRIES = 5
+
+if __name__ == '__main__':
+    url, doc_ids = sys.argv[1], sys.argv[1:]
```
Comment (Purg) [inline bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py]: Why does this take ``doc_ids`` if its never used?

File: bin/memex/hackathon_2016_07/scripts/get_parent_documents.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        .using(es) \
+        .filter(Q('match',
+                  obj_parent=json.dumps(obj_parents_l)))\
+        .scan()
+
+if __name__ == '__main__':
+    config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                               'es_config.json')
+    if not os.path.isfile(config_file):
+        sys.exit(1)
```
Comment (Purg) [inline bin/memex/hackathon_2016_07/scripts/get_parent_documents.py]: Some kind of ``print "error: ..."`` would be useful here.

File: bin/memex/hackathon_2016_07/notebooks/CP1 Data Curation.ipynb [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Curate a standard CSV for Paul\n",
```
Comment (Purg) [inline bin/memex/hackathon_2016_07/notebooks/CP1 Data Curation.ipynb]: "for Paul" isn't really necessary

File: bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py (hunk 2 above)
Comment (danlamanna) [inline bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py]: Despite this being in the SMQTK repo, I would rather not import smqtk unless it's really needed.

Comment (Purg) [inline bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py]: I suppose. It is a minor function.

**Comment by Purg** [comment]
👍

**Comment by Purg** [comment]
Oh, yeah, release notes. Please add a short description of added stuff to ``docs/release_notes/v0.7.0.md``

**Comment by Purg** [comment]
Going to need a rebase or master-merge.

File: bin/memex/hackathon_2016_07/notebooks/CP1 Data Curation.ipynb [hunk 6]
```diff
@@ -16,7 +16,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 34,
```
Comment (Purg) [inline bin/memex/hackathon_2016_07/notebooks/CP1 Data Curation.ipynb]: As possible sanitation for the future, I try to clear all cell outputs before saving them for check in for git.  This should prevent superfluous updates to run counts and output results.

**Comment by Purg** [comment]
LGTM. Will merge when tests pass.

**Pull request merged**